projects/gfbio-app/src/app/oidc/oidc.component.html

```diff
@@ -0,0 +1,11 @@
+<wave-sidenav-header>Login</wave-sidenav-header>
+<ng-container *waveIfLoggedIn>
+    <div *ngIf="user">Welcome {{ user.realName }}!</div>
+    <button mat-raised-button color="primary" (click)="logout()">Logout</button>
+</ng-container>
+<ng-container *waveIfGuest>
+    <div>Please login</div>
+    <div *ngIf="user === undefined">Undefined</div>
+    <div *ngIf="user !== undefined && user.isGuest">Guest User</div>
+    <button mat-raised-button color="primary" (click)="login()">Login with GFBio</button>
+</ng-container>
```

Comment:
Maybe, when you add some margin between "Welcome…" and the button, this component looks okay (80:20 wise).

It should not say
```
Please login
Undefined
```

Maybe add a sentence that you will be redirected to a SSO site; and hide the undefined variable.

Reply:
d0e929eb0b9cc0d3e987c9dd16658a4479d3f7c8